tools/clone.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+
+REPOSRC=$2
+LOCALREPO=$3
+BRANCH=$1
+
+# We do it this way so that we can abstract if from just git later on
```

Comment:
@filippomc not sure if its a "wise" idea to share the same git repo between different "apps"

f.e. nwbexplorer and netpyne both need repo "abc"
but netpyne needs branch "xyz"
and nwbe needs branch "qwe"

then there can be a situation where both builds run at the same time and then it's uncertain which branch is used by what build

my suggestion would be to clone the repo in a "app" specific folder instead of in a global repo folder

what do you think?

Reply:
@zsinnema agree and that's exactly what's happening, might not be clear from the code.
For example, this is how it looks like in codefresh:

```yaml
steps:
  main_clone:
    title: Clone main repository
    type: git-clone
    stage: prepare
    repo: '${{CF_REPO_OWNER}}/${{CF_REPO_NAME}}'
    revision: '${{CF_BRANCH}}'
    git: github
  post_main_clone:
    title: Post main clone
    type: parallel
    stage: prepare
    steps:
    - title: Cloning cloud-harness repository...
      type: git-clone
      stage: prepare
      repo: https://github.com/MetaCell/cloud-harness.git
      revision: '${{CLOUDHARNESS_BRANCH}}'
      working_directory: .
      git: github
    - title: Cloning nwb-explorer.git repository...
      type: git-clone
      repo: https://github.com/MetaCell/nwb-explorer.git
      revision: master
      working_directory: /home/user/OSBv2/cloud-harness/./applications/dependencies/nwb-explorer.git
      git: github
```

And in Skaffold:
```yaml
  - context: /home/user/OSBv2/cloud-harness/./applications/samples
    docker:
      buildArgs: {NOCACHE: '1706717709.5154254', REGISTRY: '', TAG: v0.2}
      dockerfile: applications/samples/Dockerfile
      ssh: default
    hooks:
      before:
      - command: [sh, tools/clone.sh, master, 'https://github.com/MetaCell/nwb-explorer.git',
          /home/user/OSBv2/cloud-harness/./applications/samples/dependencies/nwb-explorer.git]
```

Reply:
@filippomc okay cool I understand.

Then my question changes: do we really want to "cache" the git repo in the app folder? or do we clean afterwards/before and do always a `git export`

the advantage is that the build can not change the branch by executing a "git checkout <new branch>"

so a thought

Reply:
I like the idea to keep the directory as I expect development to be ongoing there too: if the codebase were settled I'd rather go with a static dockerfile extension rather than this dynamic git clone

Reply:
@filippomc I think I was not clear enough.

check this case:
* we use the script to clone a repo for example branch `develop`
* then in the Dockerfile we cd into the folder of this git repo
* then in the Dockerfile we do a `git checkout main`
resulting in a different branch than it was set in the dependencies

imo this should be prevented
easy solution is to `git export` instead of `git clone`